aicsimageio/readers/default_reader.py

```diff
@@ -1,89 +1,438 @@
+
+                # Get basic metadata
+                metadata = reader.get_meta_data()
+
+                # Create extra metadata from assumptions based off image data
+                dims, coords = self._unpack_dims_and_coords(image_data, metadata)
+
+                return xr.DataArray(
+                    image_data,
+                    dims=dims,
+                    coords=coords,
+                    attrs=metadata,
+                )
+
+    def _read_immediate(self) -> xr.DataArray:
+        """
+        Construct the in-memory xarray DataArray object for the image.
+
+        Returns
+        -------
+        image: xr.DataArray
+            The fully constructed and fully read into memory image as a DataArray
+            object. Metadata is attached in some cases as coords, dims, and attrs.
+
+        Raises
+        ------
+        exceptions.UnsupportedFileFormatError: The file could not be read or is not
+            supported.
+        """
+        # Read image
+        with self.fs.open(self.path) as open_resource:
+            reader = imageio.get_reader(
+                open_resource, format=self.extension, mode=self.imageio_read_mode
+            )
+
+            # Store image length
+            image_length = self._get_image_length(
+                fs=self.fs,
+                path=self.path,
+                format=self.extension,
+                mode=self.imageio_read_mode,
             )
 
-        # Handle single-image formats like png, jpeg, etc
-        if image_length == 1:
-            return reader.get_data(0)
+            # Handle single-image formats like png, jpeg, etc
+            if image_length == 1:
+                image_data = reader.get_data(0)
+
+            # Handle many image formats like gif, mp4, etc
+            elif image_length > 1:
+                # Read and stack all frames
+                frames = []
+                for frame in reader:
+                    frames.append(frame)
+
+                image_data = np.stack(frames)
+
+            # Get basic metadata
+            metadata = reader.get_meta_data()
+
+            # Create extra metadata from assumptions based off image data
+            dims, coords = self._unpack_dims_and_coords(image_data, metadata)
+
+            return xr.DataArray(
+                image_data,
+                dims=dims,
+                coords=coords,
+                attrs=metadata,
+            )
 
-        # Handle many image formats like gif, mp4, etc
-        elif image_length > 1:
-            frames = []
-            for i, frame in enumerate(reader.iter_data()):
-                frames.append(frame)
+    @property
+    def metadata(self):
+        if self._metadata is None:
+            self._metadata = self.xarray_dask_data.attrs
 
-            return np.stack(frames)
+        return self._metadata
 
-        raise TypeError(path)
+    @property
+    def physical_pixel_sizes(self) -> PhysicalPixelSizes:
```

Comment:
A comment or docstring defining the units of physical_pixel_sizes would be very handy!

Reply:
Docstring is inheritted from abstract base class, can see base class docstring [here](https://github.com/AllenCellModeling/aicsimageio/blob/feature/add-default-reader/aicsimageio/readers/reader.py#L464)

Reply:
Ohhhh, the _units_. Uhhhh we technically don't handle units just yet.

Reply:
So want me to add a comment to the ABC docstring about "we do not return units, check the metadata for the unit definition" @toloudis I would still argue that in 4.2 or similar there is value in adding `pint` as a unit handler.